cloudinit/distros/__init__.py

```diff
@@ -845,7 +848,13 @@ def _get_package_mirror_info(mirror_info, data_source=None,
         # ec2 availability zones are named cc-direction-[0-9][a-d] (us-east-1b)
         # the region is us-east-1. so region = az[0:-1]
         if _EC2_AZ_RE.match(data_source.availability_zone):
-            subst['ec2_region'] = "%s" % data_source.availability_zone[0:-1]
+            ec2_region = data_source.availability_zone[0:-1]
+
+            if (ALLOW_EC2_MIRRORS_ON_NON_AWS_INSTANCE_TYPES
+                    and data_source.platform_type == "ec2"):
+                subst['ec2_region'] = "%s" % ec2_region
+            elif not ALLOW_EC2_MIRRORS_ON_NON_AWS_INSTANCE_TYPES:
```

Comment:
This elif is incorrect isn't it? We only want to set subst[ec2_region'] if any([ data_source.platform_type == "ec2", ALLOW_EC2_MIRRORS_ON_NON_AWS_INSTANCE_TYPES] ?

Reply:
Yes, it is incorrect :facepalm: I will update the logic to fix this